circuits/src/zk_circuits/proof_linking.rs

```diff
@@ -0,0 +1,218 @@
+) -> Result<PlonkLinkProof, ProverError>
+where
+    [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
+{
+    // Get the group layout for the reblind <-> commitments link group
+    #[rustfmt::skip]
+    let layout =
+        get_reblind_commitments_group_layout::<MAX_BALANCES, MAX_ORDERS, MAX_FEES, MERKLE_HEIGHT>()?;
+    let pk = ValidReblind::<MAX_BALANCES, MAX_ORDERS, MAX_FEES, MERKLE_HEIGHT>::proving_key();
+
+    PlonkKzgSnark::link_proofs::<SolidityTranscript>(
+        reblind_link_hint,
+        commitments_link_hint,
+        &layout,
+        &pk.commit_key,
+    )
+    .map_err(ProverError::Plonk)
+}
+
+/// Validate a link between a proof of VALID COMMITMENTS with a proof of VALID
+/// REBLIND
+pub fn validate_commitments_reblind_link<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+    const MERKLE_HEIGHT: usize,
+>(
+    link_proof: &PlonkLinkProof,
+    reblind_proof: &PlonkProof,
+    commitments_proof: &PlonkProof,
+) -> Result<(), ProverError>
+where
+    [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
+{
+    // Get the group layout for the reblind <-> commitments link group
+    #[rustfmt::skip]
+    let layout =
+        get_reblind_commitments_group_layout::<MAX_BALANCES, MAX_ORDERS, MAX_FEES, MERKLE_HEIGHT>()?;
+    let vk = ValidReblind::<MAX_BALANCES, MAX_ORDERS, MAX_FEES, MERKLE_HEIGHT>::verifying_key();
+
+    PlonkKzgSnark::verify_link_proof::<SolidityTranscript>(
+        reblind_proof,
+        commitments_proof,
+        link_proof,
+        &layout,
+        &vk.open_key,
+    )
+    .map_err(ProverError::Plonk)
+}
+
+/// Get the group layout for the reblind <-> commitments link group
+fn get_reblind_commitments_group_layout<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+    const MERKLE_HEIGHT: usize,
+>() -> Result<GroupLayout, ProverError>
+where
+    [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
+{
+    let circuit_layout =
+        ValidReblind::<MAX_BALANCES, MAX_ORDERS, MAX_FEES, MERKLE_HEIGHT>::get_circuit_layout()
+            .map_err(ProverError::Plonk)?;
+    Ok(circuit_layout.get_group_layout(VALID_REBLIND_COMMITMENTS_LINK))
+}
+
+// --------------------------------
+// | Commitments <-> Match Settle |
+// --------------------------------
+
+/// Link a proof of VALID COMMITMENTS with a proof of MATCH SETTLE using the
+/// system wide sizing constants
+pub fn link_sized_commitments_match_settle(
+    party_id: PartyId,
+    commitments_link_hint: &MpcProofLinkingHint,
+    match_settle_link_hint: &MpcProofLinkingHint,
+    fabric: &Fabric,
+) -> Result<MpcPlonkLinkProof, ProverError> {
+    link_commitments_match_settle::<MAX_BALANCES, MAX_ORDERS, MAX_FEES>(
+        party_id,
+        commitments_link_hint,
+        match_settle_link_hint,
+        fabric,
+    )
+}
+
+/// Link a proof of MATCH SETTLE with a proof of VALID COMMITMENTS
+pub fn link_commitments_match_settle<
```

Comment:
Should we make an analogue of this method for the singleprover (internal match) setting? We'd just use `ProofLinkingHint`s and no `Fabric` necessary right

Reply:
good call out, will do